src/datasets/upload/image.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  Input as GrpcInput,
+  Annotation,
+} from "clarifai-nodejs-grpc/proto/clarifai/api/resources_pb";
+import { ClarifaiDataLoader, ClarifaiDataset } from "./base";
+import path from "path";
+import { v4 as uuid } from "uuid";
+import { Input } from "../../client/input";
+import {
+  VisualClassificationFeatures,
+  VisualDetectionFeatures,
+  VisualSegmentationFeatures,
+} from "./features";
+import { JavaScriptValue } from "google-protobuf/google/protobuf/struct_pb";
+
+export class VisualClassificationDataset extends ClarifaiDataset<ClarifaiDataLoader> {
+  constructor(args: { dataGenerator: ClarifaiDataLoader; datasetId: string }) {
+    super(args);
+  }
+
+  protected extractProtos({
+    batchInputIds,
+  }: {
+    batchInputIds: string[];
+  }): [GrpcInput[], Annotation[]] {
+    const inputProtos: GrpcInput[] = [];
+    const annotationProtos: Annotation[] = [];
+
+    const processDataItem = (id: string) => {
+      const dataItem = this.dataGenerator.getItem(
+        Number(id),
+      ) as VisualClassificationFeatures;
+      let metadata: Record<string, JavaScriptValue> = {};
+      const imagePath = dataItem.imagePath;
+      const labels = Array.isArray(dataItem.labels)
+        ? dataItem.labels.map((label) => label.toString())
+        : [(dataItem.labels as string).toString()]; // clarifai concept expects labels to be an array
+      const inputId = `${this.datasetId}-${String(dataItem.id)}`;
+      const geoInfo = dataItem.geoInfo;
+
+      if (dataItem.metadata) {
+        metadata = dataItem.metadata;
+      } else if (imagePath) {
+        metadata = {
+          filename: path.basename(imagePath),
+        };
+      }
+
+      this.allInputIds[id] = inputId;
+
+      if (dataItem.imageBytes) {
+        inputProtos.push(
+          Input.getInputFromBytes({
+            inputId,
+            imageBytes: dataItem.imageBytes,
+            datasetId: this.datasetId,
+            labels,
+            geoInfo: {
+              latitude: geoInfo?.[0] as number,
+              longitude: geoInfo?.[1] as number,
+            },
+            metadata,
+          }),
+        );
+      } else {
+        inputProtos.push(
+          Input.getInputFromFile({
```

Comment:
I think static constructors like this should not start with `get` because that implies we're GETting that input from the API. Maybe something like `Input.fromFile(...)` would be better. Same with `fromDirectory` etc.

Reply:
We decided to keep the naming convention to be same across all our SDKs, so we will update it across all the SDKs in the future.